### DISCUSSION TITLE
this is keycloak added to the existing auth or only keycloak? what is with this deno?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [nordeck/jitsi-keycloak-adapter#25] [General]
I would like to add keycloak sso as an extra option, not the only one. Is this manual doing this?

https://github.com/nordeck/jitsi-keycloak-adapter/blob/main/docs/setup-standalone.md

Why is it necessary to have this deno? If you add a wordpress plugin you do not need to have a daemon running.

**Comment by emrahcom** [comment]
Could you tell me more about what you plan to do?

This adapter is to add Keycloak as an authentication option to [Jitsi](https://jitsi.org/). Is this what you want?

**Reply by f1-outsourcing** [reply]
I would like to keep my regular login with user + passwd and a button sso (or auto detection) for a detected openid/saml session.

**Reply by emrahcom** [reply]
I'm still not sure if you are talking about Jitsi or another software (maybe a site based on Wordpress). Could you clarify this please?

**Reply by f1-outsourcing** [reply]
Yes of course I like to add it to jitsi. :) I am currently looking at this prosody because that does auth for jitsi, I am having ldap there, but I also would like to keep that as backup. I have almost finished, but looks like it is either this
`authentication = "ldap";`
or this
`authentication = "token";`

**Reply by emrahcom** [reply]
There is no ready-made solution for this use case. In summary, you should:

- configure `token` authentication in Prosody
- redirect participants to your custom login page before meeting
- this page should provide options based on LDAP and SSO
- if everything goes well on this custom authentication page, it should provide a token for the participant and redirect her to Jitsi with a token

**Reply by emrahcom** [reply]
Another option is to handle LDAP in Keycloak and using this adapter

**Reply by f1-outsourcing** [reply]
I have actually, but it is limited to a subset and as I running into all kinds of bugs with different applications I prefer to use both.
